app/src/containers/home/redux/addPostToDatabaseReducer.js

```diff
@@ -0,0 +1,57 @@
+import Immutable from 'seamless-immutable';
+
+/* ------------- Types and Action Creators ------------- */
+
+export const AddPostToDatabaseTypes = {
+  ADD_POST_TO_DATABASE_REQUEST: 'ADD_POST_TO_DATABASE_REQUEST',
+  ADD_POST_TO_DATABASE_REQUEST_SUCCESS: 'ADD_POST_TO_DATABASE_REQUEST_SUCCESS',
+  ADD_POST_TO_DATABASE_REQUEST_FAILURE: 'ADD_POST_TO_DATABASE_REQUEST_FAILURE',
+};
+
+const Types = AddPostToDatabaseTypes;
+export const AddPostToDatabaseCreators = {
+  addPostToDatabaseRequest: payload => ({
+    type: Types.ADD_POST_TO_DATABASE_REQUEST,
+    payload,
+  }),
+  addPostToDatabaseRequestSuccess: post => ({
+    type: Types.ADD_POST_TO_DATABASE_REQUEST_SUCCESS,
+    post,
+  }),
+  addPostToDatabaseRequestFailure: error => ({
+    type: Types.ADD_POST_TO_DATABASE_REQUEST_FAILURE,
+    error,
+  }),
+};
+
+/* ------------- Initial State ------------- */
+
+const INITIAL_STATE = Immutable({
+  error: null,
+  fetching: false,
+  success: false,
+  user: null,
```

Comment:
It should be `post` instead of `user`
```suggestion
  post: null,
```